tests/suite/test_v_s_route_grpc.py

```diff
@@ -0,0 +1,201 @@
+                        ingress_controller_prerequisites.config_map['metadata']['name'],
+                        ingress_controller_prerequisites.namespace,
+                        f"{DEPLOYMENTS}/common/nginx-config.yaml")
+        delete_common_app(kube_apis, app_name, test_namespace)
+        pytest.fail(f"VSR GRPC setup failed")
+
+    def fin():
+        print("Clean up:")
+        replace_configmap_from_yaml(kube_apis.v1,
+                        ingress_controller_prerequisites.config_map['metadata']['name'],
+                        ingress_controller_prerequisites.namespace,
+                        f"{DEPLOYMENTS}/common/nginx-config.yaml")
+        delete_common_app(kube_apis, app_name, test_namespace)
+
+    request.addfinalizer(fin)
+
+@pytest.mark.vsr
+@pytest.mark.smoke
+@pytest.mark.parametrize('crd_ingress_controller, v_s_route_setup',
+                         [({"type": "complete", "extra_args": [f"-enable-custom-resources"]},
+                           {"example": "virtual-server-route-grpc"})],
+                         indirect=True)
+class TestVirtualServerRouteGrpc:
+
+    def patch_valid_vs_route(self, kube_apis, v_s_route_setup) -> None:
+        """
+        Function to revert vs deployment to valid state
+        """
+        patch_src = f"{TEST_DATA}/virtual-server-route-grpc/route-multiple.yaml"
+        patch_v_s_route_from_yaml(
+            kube_apis.custom_objects,
+            v_s_route_setup.route_m.name,
+            patch_src,
+            v_s_route_setup.route_m.namespace,
+        )
+
+    def deploy_tls_secrets(self, kube_apis, v_s_route_setup) -> None:
+        """
+        Function to deploy secrets to the vs route namespaces.
+        """
+        print("------------------------- Deploy Secrets -----------------------------")
+        src_sec_yaml = f"{TEST_DATA}/virtual-server-route-grpc/tls-secret.yaml"
+        create_secret_from_yaml(kube_apis.v1, v_s_route_setup.route_m.namespace, src_sec_yaml)
+        create_secret_from_yaml(kube_apis.v1, v_s_route_setup.route_s.namespace, src_sec_yaml)
+        wait_before_test(1)
+
+    @pytest.mark.parametrize("backend_setup", [{"app_type": "grpc-vs"}], indirect=True)
+    def test_config_after_setup(self, kube_apis, ingress_controller_prerequisites, crd_ingress_controller, 
+                                backend_setup, v_s_route_setup):
+        self.deploy_tls_secrets(kube_apis, v_s_route_setup)
+        print("\nStep 1: assert config")
+        ic_pod_name = get_first_pod_name(kube_apis.v1, ingress_controller_prerequisites.namespace)
+        config = get_vs_nginx_template_conf(kube_apis.v1,
+                                            v_s_route_setup.namespace,
+                                            v_s_route_setup.vs_name,
+                                            ic_pod_name,
+                                            ingress_controller_prerequisites.namespace)
+
+        assert "proxy_connect_timeout 60s;" not in config
```

Comment:
these assertions can be separated out in custom_assertions.py and can be reused in all 3 tests by a single function call.

Reply:
Done